### ISSUE TITLE
Talk me through the whole "Hard limit" thing...

### ISSUE THREAD

**Issue by autoSteve** [BJReplay/ha-solcast-solar#109]
A configurable inverter "hard limit" is available in the integration, and its documentation is poor to non-existant. (Thank you for the prod on this, @ProphetOfDoom.)

I really don't understand its practical implementation, so if someone knows why it would be employed, or in fact use this feature then please feel free to raise a readme PR.

This hard limit is applied to the overall sites total expected output, so that will work. Where it currently will not work is for the breakdown of site forecast values in attributes. To make that work I need to know what it is, and why it is.

Thanks in advance for any information, suggestions, PRs, or anything really...

**Comment by autoSteve** [comment]
@CarrapiettM, any information from you? This was an Oziee v4.0.22 addition, just before he..., well, @BJReplay assumed the drivers seat for this integration.

**Comment by CarrapiettM** [comment]
> @CarrapiettM, any information from you? This was an Oziee v4.0.22 addition, just before he..., well, @BJReplay assumed the drivers seat for this integration.

Yeah , the hard limit was put in place as I have a dual array and solcast was predicting over my 6kw inverter limit, adjusting the limits in solcast was not helping, hence the hard limit was built in.

**Comment by CarrapiettM** [comment]
> That is brilliant, @CarrapiettM.
> 
> To document this in the readme, I take it that you have over-sized strings pointing different directions, and feeding a single inverter, yeah? It's the over-sized that I think matters, correct? It'd be ace to see your Solcast sites config for DC/AC generation.
> 
> For the code that currently will not work (the sites breakdown in attributes) my guess would be to proportion the hard limit based on expected generation for any given half-hour period.

Is this what you are after ?

<img width="156" alt="image" src="https://github.com/user-attachments/assets/fb9c3cb4-e862-4b04-b6d2-7f6ced07bc82">
<img width="157" alt="image" src="https://github.com/user-attachments/assets/177e41d5-8374-4f38-af64-882bbd997b94">

**Comment by autoSteve** [comment]
Righto. So your DC potential generation is 5.5kW max for each string (total max 11.0 if the sun were inclined to split itself in two at the right positions), which I suppose would be considered "over-sized" for the inverter. I take it that sometimes the DC generation well exceeds 5.5kW, like in summer with the sun overhead.

This makes sense, as limiting the AC capacity in Solcast for this case would break my head.

**Comment by autoSteve** [comment]
A review of my doco commit would be most welcome, @CarrapiettM.

**Comment by autoSteve** [comment]
Many thanks, @CarrapiettM.

**Comment by autoSteve** [comment]
> Sorry for the brain dump

Mind not blown, @ProphetOfDoom. But a cool explanation of things you wanted to know about PV generation, but were too afraid to ask...

On topic, I think I've summarised the hard limit reasonably, as it relates to two sites facing different directions, yet connected to a single string inverter. It may be a reasonabe summary. Maybe.

I got micro inverters here, so I got nothin', really.
